File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
#include <string.h>


int main (int argc, char** argv)
{
    int k;
 if (argc!=2) //proverka na k-vo argv
 {
   printf("Usage: ./caesar k\n");
   return 1;
 }
 
if (((int)(argv[1]))<0)
{
  printf("Vedene vid'emne chislo"); 
  return 1;
}

if (atoi(argv[1])>26)  //esli >26 -> vuchislayem poziciu
     k=(int)atoi(argv[1])%26;
else
     k = (int)atoi(argv[1]);
     

printf("Plaintext:   ");
string str=get_string();

for(int i=0;i<strlen(str);i++)
{
  
  if ((str[i]>='a')&&(str[i]<='z'))
  {
     if ((str[i]+k)>'z')
         str[i]=(int)('a')+(k-((int)('z')-str[i]))-1; //esli zakonchilsya alfavit -> nachinaem snachala shitat'
     else   
         str[i]+=k;
  }
  
   if ((str[i]>='A')&&(str[i]<='Z'))
  {
     if ((str[i]+k)>'Z')
         str[i]=(int)('A')+(k-((int)('Z')-str[i]))-1;//esli zakonchilsya alfavit -> nachinaem snachala shitat'
     else
         str[i]+=k;
  }
}

printf("ciphertext:  %s\n",str);
return 0;
}

File: pset6/caesar.py
import sys

num=0
if __name__ == "__main__":
    for param in sys.argv:
        num+=1

    if len(sys.argv)!=2:
        print("Usage: ./caesar k");
        sys.exit(1)
   
    if ((int(sys.argv[1]))<0):
        print("Vedene vid'emne chislo") 
    
    if (int(sys.argv[1])>26): 
        k=int(sys.argv[1])%26;
    else:
        k = int(sys.argv[1])
  
str=list(input("plaintext:  "))
i=0

while i<len(str):
    if ((ord(str[i])>=ord('a'))and(ord(str[i])<=ord('z'))):
        if ((ord(str[i])+k)>ord('z')):
            str[i]=chr(ord('a')+(k-(ord('z')-ord(str[i])))-1)
        else:   
            str[i]=chr(ord(str[i])+k)
 
  
    if ((ord(str[i])>=ord('A'))and(ord(str[i])<=ord('Z'))):
        if ((ord(str[i])+k)>ord('Z')):
            str[i]=chr(ord('A')+(k-(ord('Z')-ord(str[i])))-1)
        else:
            str[i]=chr(ord(str[i])+k)
    i+=1
str1="".join(str)
print("ciphertext:",str1);